### ISSUE TITLE
🐛 Bug Report: @novu/framework: 2.4.0 Breaking imports

### ISSUE THREAD

**Issue by tmanick01** [notifirehq/notifire#6954]
### 📜 Description

version 2.4.0 no longer exports the type Workflow, causing imports to break in typescript

### 👟 Reproduction steps

install latest @novu/framework
Try importinng the Workflow type

### 👍 Expected behavior

Type should be exported and should be accessible in client code, where type inference may not be straight forward, for example in a custom monorepo

### 👎 Actual Behavior with Screenshots

![image](https://github.com/user-attachments/assets/614d4cd1-5934-4908-947d-43f72fffcd1b)

### Novu version

Novu SaaS

### npm version

9.5.1

### node version

18.16.0

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find a similar issue

### 🏢 Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/novuhq/novu/blob/main/CONTRIBUTING.md)

### Are you willing to submit PR?

None

**Comment by rifont** [comment]
Firstly, apologies for this disruption!

Version `2.4.0` ships with a reduced public API focussed on improved DX with less noise. Specifically, we removed a lot of types from the public `@novu/framework` entry point. The `Workflow` type was not documented publicly on `docs.novu.co` and as such, this change was considered non-breaking.

We have an open PR (https://github.com/novuhq/novu/pull/6812) that will ship soon, providing `InferWorkflowPayload` and `InferWorkflowId` type inference utilities and re-exporting the `Workflow` type for user-land usage. At the same time we will publish docs.novu.co documentation for these types.

Could you please share your use-case for accessing the `Workflow` type? It will be useful for us to understand so we can create more useful type inference utilities in future iterations 🙏

In the interim, please import change your `Workflow` import to use the `internal` API:
```typescript
import { Workflow } from `@novu/framework/internal`;
```

It's important to note that the `@novu/framework/internal` entry point is not stable and is subject to breaking changes.

**Comment by tmanick01** [comment]
Thank you for the response @rifont. We have a plugin system and this error happens in the plugin we wrote for novu. The Workflow type is used in the interface declaration for that Plugin. We were able to work around that by using ReturnType<type of workflow>, which solved the build failure.
Howevere another breaking behavior is that the sync via the cli seems to be broken : 
![image](https://github.com/user-attachments/assets/28c57b38-4dc8-42bc-802e-c322bb60194d)
trying the to test the workflow from the dashboard has the same error as well: 

`
{
  "statusCode": 500,
  "retryCount": 2,
  "message": "Internal Server Error",
  "raw": {
    "message": "Unknown BridgeError: crypto is not defined",
    "data": {
      "stack": "ReferenceError: crypto is not defined\n    at Or (/usr/src/ms-notification/node_modules/@novu/framework/dist/cjs/servers/express.cjs:1:11777)\n    at bt.validateHmac (/usr/src/ms-notification/node_modules/@novu/framework/dist/cjs/servers/express.cjs:95:6587)\n    at bt.handleAction (/usr/src/ms-notification/node_modules/@novu/framework/dist/cjs/servers/express.cjs:95:4619)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async /usr/src/ms-notification/node_modules/@novu/framework/dist/cjs/servers/express.cjs:95:3618"
    },
    "code": "BridgeError"
  }
}
`

Reverting @novu/framework: 2.3.0 fixes this, and everything works as before
![image](https://github.com/user-attachments/assets/d26de638-4658-4cb6-83c0-0b915ef57a16)

**Comment by rifont** [comment]
Thank you for sharing more information about your setup.

In `2.4.0` we replaced the internal usage of `node:crypto` module with the global Web `crypto` API to support edge runtimes like `Cloudflare`, which don't have access to Node APIs.

It appears that your runtime environment doesn't make that global `crypto` API available. Could you please share information about your runtime environment? The following will be helpful:
* Runtime environment (e.g. Node, Cloudflare, Deno)
* Runtime version

**Comment by rifont** [comment]
Never mind the above, I see you are using Node 18 from the original description. Please raise a separate issue for the `crypto` issue and we'll assist there.

**Comment by rifont** [comment]
I raised a new issue (https://github.com/novuhq/novu/issues/6982) for the `crypto` error.

**Comment by tmanick01** [comment]
Thank you @rifont , got side tracked with other  priorities. Will add more details in the new issue soon.

**Comment by rifont** [comment]
The `Workflow` type is now available in   `@novu/framework`@[2.5.0](https://www.npmjs.com/package/@novu/framework/v/2.5.0) 🎉

```typescript
import { Workflow } from '@novu/framework';
```
